cpu/cpupower_monitor.py

```diff
@@ -0,0 +1,118 @@
+                self.states_list = (line.split('Available idle states: ')[-1]).split()
+                break
+        self.log.info("Idle states on the system are: ", self.states_list)
+        self.run_cmd_out("cpupower monitor")
+
+    def run_cmd_out(self, cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True,
+                                     sudo=True).decode("utf-8")
+
+    def check_zero_nonzero(self, cmd, stop_state_index):
+        output = self.run_cmd_out(cmd)
+        values_list = []
+        split_index = 2
+        if 'WARNING' in output:
+            split_index = 3
+        for line in output.splitlines()[split_index:]:
+            stop_state_values = line.split('|')
+            values_list.append(stop_state_values[stop_state_index].strip())
+        for value in values_list:
+            if float(value) != 0.00:
+                return 1
+        return 0
+
+    def test_workload(self):
+        tarball = self.fetch_asset('http://sourceforge.net/projects/ebizzy/'
+                                   'files/ebizzy/0.3/ebizzy-0.3.tar.gz')
+        archive.extract(tarball, self.workdir)
+        self.sourcedir = os.path.join(self.workdir, 'ebizzy-0.3')
+        os.chdir(self.sourcedir)
+        process.run('./configure', shell=True)
+        build.make(self.sourcedir)
+        self.run_cmd_out("cpupower monitor")
+        self.log.info("============ Starting ebizzy tests ==============")
+        obj = process.SubProcess('./ebizzy &', verbose=False,
+                                 shell=True)
+        obj.start()
+        time.sleep(2)
+        for i in range(len(self.states_list)):
+            zero_nonzero = self.check_zero_nonzero("cpupower monitor", i + 1)
+            if zero_nonzero:
+                self.fail("cpus entered idle states during ebizzy workload")
+            self.log.info("no cpus entered idle states while running ebizzy")
+        time.sleep(10)
+        zero_nonzero = 0
+        for i in range(len(self.states_list)):
+            zero_nonzero = zero_nonzero + self.check_zero_nonzero("cpupower monitor", i + 1)
+        if not zero_nonzero:
+            self.fail("cpus have not entered idle states after killing ebizzy workload")
+        self.log.info("cpus have entered idle states after killing work load")
+
+    def test_disable_idlestate(self):
+        for i in range(len(self.states_list)):
+            process.run('cpupower -c all idle-set -d %s' % i, shell=True)
+            time.sleep(5)
+            zero_nonzero = self.check_zero_nonzero("cpupower monitor", i + 1)
+            if zero_nonzero:
+                self.fail("cpus have entered the disabled idle states.")
+            self.log.info("cpus have not entered disabled idle states")
+            process.run('cpupower -c all idle-set -E', shell=True)
+
+    @skipIf("powerpc" not in cpu.get_arch(), "Skip, SMT specific tests")
+    def test_idlestate_smt(self):
+        process.run('ppc64_cpu --smt=off', shell=True)
```

Comment:
@Pavithra1602 Sorry for the delayed comment. Can this test be changed to run in a loop with list of smt values `['off', '2', '4', 'on']` to avoid repetition?